### DISCUSSION TITLE
Check if the field exists

### DISCUSSION THREAD

**Discussion by fabioselau077** [firebase/flutterfire#7333] [Feature request]
It would be nice to have a function that checks if the field exists in that document.
Because currently I added a new field in the new documents registered in Firestore but when I try to list and there is an old document (without the new field) it returns "bad state field does not exist within the documentsnapshot platform".
But I can't do anything else, because if I try to create a check for when this field is null to have a default value I just return an error because to retrieve I need to use it as an array and as there's no such field, I get no output.

**Comment by japarico** [comment]
I wonder if there is a solution to that too.
